TYPO3.Flow/Tests/Unit/Http/Component/TrustedProxiesComponentTest.php

```diff
@@ -339,6 +339,46 @@ public function hostIsNotOverridenIfTheHeaderIsNotTrusted()
     }
 
     /**
+     * @test
+     */
+    public function hostIsOverridenIfTheHeaderIsTrusted()
+    {
+        $request = Request::create(new Uri('http://acme.com'), 'GET', array(), array(), array('HTTP_X_FORWARDED_HOST' => 'neos.io'));
+        $trustedRequest = $this->callWithRequest($request);
+        $this->assertEquals('neos.io', $trustedRequest->getUri()->getHost());
+    }
+
+    /**
+     * @test
+     */
+    public function portIsOverridenIfTheHostHeaderContainsPort()
+    {
+        $request = Request::create(new Uri('http://acme.com'), 'GET', array(), array(), array('HTTP_X_FORWARDED_HOST' => 'neos.io:443'));
+        $trustedRequest = $this->callWithRequest($request);
+        $this->assertEquals(443, $trustedRequest->getUri()->getPort());
+    }
+
+    /**
+     * @test
+     */
+    public function portIsOverridenIfTheHostHeaderContainsPortAlsoIfProtocolHeaderIsSet()
+    {
+        $request = Request::create(new Uri('http://acme.com'), 'GET', array(), array(), array('HTTP_X_FORWARDED_HOST' => 'neos.io:443', 'HTTP_X_FORWARDED_PROTO' => 'http'));
+        $trustedRequest = $this->callWithRequest($request);
+        $this->assertEquals(443, $trustedRequest->getUri()->getPort());
+    }
+
+    /**
+     * @test
+     */
+    public function portFromHostHeaderIsOverriddenByPortHeader()
```

Comment:
Here is one theoretic scenario in which the two might be different:

Suppose you have two proxies, one that specifies the port via `X-Forwarded-Port`, the other via the `X-Forwarded-Host`. You'd end up with either `X-Forwarded-Host: acme.com, foo-bar.org:8080; X-Forwarded-Port: 443` or `X-Forwarded-Host: acme.com:443, foo-bar.org; X-Forwarded-Port: 8080`.

Now of course that also doesn't make clear which is "correct" generally (I'm not even sure that would be handled sanely - curse you `mod_proxy`!).